src/fluree/db/query/json_ld/response.cljc

```diff
@@ -15,12 +15,13 @@
 
 (defn wildcard-spec
   [db cache compact-fn pid]
-  (or (get @cache pid)
-      (let [p-spec (if-let [spec (get-in db [:schema :pred pid])]
-                     (assoc spec :as (compact-fn (:iri spec)))
-                     {:as pid})]
-        (vswap! cache assoc pid p-spec)
-        p-spec)))
+  (go-try
```

Comment:
If this isn't in a high frequency code path it may not matter, but generally when we use a cache for speed, we want to check the cache before spawning core/async chans which is expensive at volume.

Here you spawn a chan, then conditionally check for an exception, then check for a cached value - where the cached value check needs not a chan, nor an exception check.

You'll note the other similar patterns here where cache checking is done before spawning a chan (see line 68 below)

Reply:
Good catch, this is in a high frequency code path. I've refactored it to only spawn the thread as a last resort.